integrations/visual_studio/creating_packages.rst

```diff
@@ -0,0 +1,228 @@
+.. _visual_studio_packages:
+
+|visual_logo| Creating and reusing packages with Visual Studio
+==================================================================
+
+Conan has different helpers to manage Visual Studio and MSBuild based projects.
+This how-to illustrates how to put them together to create and consume packages that are purely
+based on Visual Studio. This how-to is using VS2015, but other versions can be used too.
+
+
+Creating packages
+------------------
+
+Start cloning the existing example repository, containing a simple "Hello World" library, and
+application:
+
+.. code-block:: bash
+
+    $ git clone https://github.com/conan-io/examples
+    $ cd features/visual_studio/creating_packages
+
+It contains a ``src`` folder with the source code and a ``build`` folder with a Visual Studio 2015
+solution, containing 2 projects: the HelloLib static library, and the Greet application. Open it:
+
+::
+
+    build\HelloLib\HelloLib.sln
+
+You should be able to select the ``Greet`` subproject -> ``Set as Startup Project``.
+Then build and run the app with Ctrl+F5. (Debug -> Start Without Debugging)
+
+
+.. code-block:: bash
+
+    $ Hello World Debug!
+    # Switch IDE to Release mode, repeat
+    $ Hello World Release!
+
+Because the ``hello.cpp`` file contains an ``#ifdef _DEBUG`` to switch between debug and release
+message.
+
+In the repository, there is already a ``conanfile.py`` recipe:
+
+.. code-block:: python
+
+    from conans import ConanFile
+    from conan.tools.microsoft import MSBuild
+
+
+    class HelloConan(ConanFile):
+        name = "Hello"
+        version = "0.1"
+        license = "MIT"
+        url = "https://github.com/conan-io/examples"
+        settings = "os", "compiler", "build_type", "arch"
+        exports_sources = "src/*", "build/*.vcxproj*", "build/*.sln*"
+
+        def build(self):
+            msbuild = MSBuild(self)
+            msbuild.build("build/HelloLib/HelloLib.sln")
+
+        def package(self):
+            self.copy("*.h", dst="include", src="src")
+            self.copy("*.lib", dst="lib", keep_path=False)
+            self.copy("*.dll", dst="bin", keep_path=False)
+
+        def package_info(self):
+            self.cpp_info.libs = ["HelloLib"]
+
+
+This recipe is using the (:ref:)`MSBuild() build helper <msbuild>` to build the ``sln`` project.
+If our recipe has ``requires``, the ``MSBUILD`` helper will also take care of inject all the needed
```

Comment:
This is wrong, the ``MSBuild`` helper doesn't inject all of that, it is mostly dummy, a thin wrapper around the command line. That responsibility is not part of ``MSBuildDeps`` and ``MSBuildToolchain`` (better leave the explanation when you use them)

Reply:
Done